internal/plugin/v1/testutils/testutils.go

```diff
@@ -16,14 +17,23 @@ type CanProvideTestCase struct {
 
 // CanProvide calls GetValues on the provider and ensures that the provider response for
 // the given id has the expected value and no error
-func CanProvide(provider plugin_v1.Provider, id string, expectedValue string) func() {
+func CanProvide(provider plugin_v1.Provider,
+	id string,
+	expectedValue string,
+	shouldError bool) func() {
 	return func() {
 		values, err := provider.GetValues(id)
 
-		convey.So(err, convey.ShouldBeNil)
-
-		value := values[id]
-		assertGoodProviderResponse(value, expectedValue)
+		if shouldError {
+			for _, response := range values {
+				convey.So(response.Error, convey.ShouldBeError,
+					errors.New(expectedValue))
+			}
+		} else {
+			convey.So(err, convey.ShouldBeNil)
+			value := values[id]
+			assertGoodProviderResponse(value, expectedValue)
+		}
```

Comment:
I think `convey.So(values[id].Error, convey.ShouldBeNil)` was missing in the previous implementation of `CanProvide`...

```suggestion
		convey.So(values[id].Error, convey.ShouldBeNil)
		convey.So(err, convey.ShouldBeNil)

		value := values[id]
		assertGoodProviderResponse(value, expectedValue)
```

Reply:
Ah very true, good call